### ISSUE TITLE
additionalData seems to be a string

### ISSUE THREAD

**Issue by beevelop** [OneSignal/OneSignal-Cordova-SDK#97]
I am currently using `onesignal-cordova-plugin@2.0.6` in an Ionic project and stumbled upon problems with using `additionalData`. It seems as if `additionalData` is not provided as an Object.

`typeof data.notification.payload.additionalData` reports a `string` and logging the value to Sentry  results in a logged string.

I "polyfilled" the functionality with the following snippet (with Lodash available), what seems to work:

``` js
if (_.has(data, 'notification.payload.additionalData') &&
  _.isString(data.notification.payload.additionalData)) {
  data.notification.payload.additionalData = JSON.parse(data.notification.payload.additionalData);
}
```

This should be handled by [src/android/com/plugin/gcm/OneSignalPush.java#L89](https://github.com/OneSignal/OneSignal-Cordova-SDK/blob/master/src/android/com/plugin/gcm/OneSignalPush.java#L89), shouldn't it?

**Comment by jkasten2** [comment]
@beevelop Sorry about the delay, I'll test this out next week and get back to you.

Thanks.

**Comment by dylanvdmerwe** [comment]
Noticed this here as well when testing on Android. The notificationOpened payload's additionalData is a string whereas it should be an object.

See attached screenshot:
![image](https://cloud.githubusercontent.com/assets/2852839/19836765/307ac302-9eb3-11e6-8004-dd0cd5027ec2.png)

**Comment by tavosansal** [comment]
Using the same version 2.0.6 and having the same issue. Sometimes I get the entire data object as a string and it is very annoying to try to figure out if I need to parse the string to JSON or not. Please advise as to how to avoid this.

**Comment by aurbina83** [comment]
tavosanal, this is what I've been doing.

```
let notificationOpenedCallback = function(data){ 
    var d;
    if(typeOf data.notification.payload.additionalData == 'string') {
        d = JSON.parse(data.notification.payload.additionalData;
    } else {
          d = data.notification.payload.additionalData;
    }

    //do stuff with d;
}
```

**Comment by jkasten2** [comment]
@aurbina83 @tavosansal @beevelop This has now been fixed in 2.0.7. Both iOS and Android will return `additionalData` as an `Object`.

**Comment by burakodabas** [comment]
**FOR typescript or ionic one signal additionalData access**
```
  this.oneSignal.handleNotificationOpened().subscribe((data) => {
    if(typeof data.notification.payload.additionalData.message !== 'undefined')
    console.log('WANT TO SAY: ' + data.notification.payload.additionalData.message );
  });
```
